src/actions/read/observation.ts

```diff
@@ -1,66 +1,60 @@
+import { DEFAULT_EPOCH_BLOCK_LENGTH } from '../../constants';
 import {
-  DEFAULT_EPOCH_BLOCK_LENGTH,
-  DEFAULT_START_HEIGHT,
-} from '../../constants';
-import { ContractReadResult, IOState, PstAction } from '../../types';
-import { getEpochStart, getPrescribedObservers } from '../../utilities';
+  getEpochBoundariesForHeight,
+  getPrescribedObserversForEpoch,
+} from '../../observers';
+import {
+  BlockHeight,
+  ContractReadResult,
+  IOState,
+  PstAction,
+} from '../../types';
 
 export const prescribedObserver = async (
   state: IOState,
   { input: { target, height } }: PstAction,
 ): Promise<ContractReadResult> => {
-  const { settings, gateways } = state;
-  if (!height) {
-    height = +SmartWeave.block.height;
-  }
-
-  const currentEpochStartHeight = getEpochStart({
-    startHeight: DEFAULT_START_HEIGHT,
-    epochBlockLength: DEFAULT_EPOCH_BLOCK_LENGTH,
-    height: height,
+  const { settings, gateways, distributions } = state;
+  const { epochStartHeight, epochEndHeight } = getEpochBoundariesForHeight({
+    currentBlockHeight: new BlockHeight(height || +SmartWeave.block.height),
+    epochZeroBlockHeight: new BlockHeight(distributions.epochZeroBlockHeight),
+    epochBlockLength: new BlockHeight(DEFAULT_EPOCH_BLOCK_LENGTH),
   });
 
-  const prescribedObservers = await getPrescribedObservers(
+  const prescribedObservers = await getPrescribedObserversForEpoch({
     gateways,
-    settings.registry.minNetworkJoinStakeAmount,
-    settings.registry.gatewayLeaveLength,
-    currentEpochStartHeight,
-  );
+    minNetworkJoinStakeAmount: settings.registry.minNetworkJoinStakeAmount,
+    epochEndHeight: epochEndHeight,
+    epochStartHeight: epochStartHeight,
+    distributions,
+  });
 
-  if (
-    prescribedObservers.some(
-      (observer) =>
-        observer.gatewayAddress === target ||
-        observer.observerAddress === target,
-    )
-  ) {
-    // The target with the specified address is found in the prescribedObservers list
-    return { result: true };
-  } else {
-    return { result: false };
-  }
+  // The target with the specified address is found in the prescribedObservers list
+  return {
+    result: prescribedObservers.some(
+      (observer) => observer.observerAddress === target,
+    ),
+  };
```

Comment:
thinking if this should expect the `target` to be the gateway that is prescribed - not the wallet. there can be one observer wallet for multiple gateways so this would return if the `gateway` address provided is a prescribed observer, rather than if the wallet address is. 🤔

Reply:
cc @vilenarios